homeassistant/components/tuya/entity.py

```diff
@@ -58,7 +58,7 @@ def step_scaled(self) -> float:
 
     def scale_value(self, value: float) -> float:
         """Scale a value."""
-        return value / (10**self.scale)
+        return value * self.step / (10**self.scale)
```

Comment:
Unfortunately, this change is not correct.

This occurrence is seen more often, and is generally caused by the device. Please contact Tuya support or the device manufacturer instead.

The currently calculation is correct (and as documented by Tuya). If we would merge this, we would break basically all other Tuya devices that have a numeric value communicated (which are a lot of them).

../Frenck

Reply:
OK, thank you for your time!

Reply:
Could the step size be made configurable somehow? The Localtuya extension supports this, but has some other issues why the core support is better option.
Apparently there's a lot of these misbehaving thermostats on the market. I wonder how the Tuyas own smart life app manages these devices.

Reply:
From IOT Tuya portal, the device debugging info says this:
`temp_set | Integer | {   "unit": "℃",   "min": 10,   "max": 70,   "scale": 1,   "step": 5 }`
i'm not a coder, but all other devices have  "Step:1"
why the "step:5" can't be used in calculation? I think that will not break existing devices.

Do "step" ever being taken into account in calc?